sdk/python/kfp/_client.py

```diff
@@ -727,6 +743,12 @@ def create_recurring_run(
     Returns:
       A Job object. Most important field is id.
     """
+    if experiment_id is None and experiment_name is None:
+      raise ValueError('Either job_id or experiment_name is required')
+    if not experiment_id:
```

Comment:
nit: `if` -> `elif`

Reply:
Yeah I think this is a nuanced one, whilst technically you're correct that both approaches would work I see this as not a valid `elif` case. The rationale is that likely in that case I'd have written the logic instead as:

```python
if not experiment_id:
  if not experiment_name:
    raise ValueError(...)
  
  experiment = self.create_experiment(...)
  ...
```

Whilst this is technically the shortest path I think the code is less clear and consistent across methods in the `client`. Really I like the idea of having the validation standalone as the first step in the method, also this happens to be consistent with how `kfp` have written it too. 

I think this is a classic case of different opinions and all are completely valid and correct, so with that in mind I'm going to resolve with no action just to better conform to the `kfp` standard really so I don't rock the boat too much when I try to get these changes accepted upstream! Thanks for suggestion though 😄